prefect_aws/ecs.py

```diff
@@ -606,13 +606,30 @@ def prepare_for_flow_run(
 
         new = super().prepare_for_flow_run(flow_run, deployment=deployment, flow=flow)
 
-        # Update ENV using flow-run details
+        # Update 'ECSTask' using flow-run context & details
+        # update 'env'
         env_update = flow_run.context.get("env", {})
         for k, v in env_update.items():
             env_update[str(k).upper()] = str(v)
-        env_update[ORCHESTRATION_PREFIX + "JOB_ID"] = str(flow_run.id)
+
+        _job_id_key = ORCHESTRATION_PREFIX + "JOB_ID"
+        if _job_id_key not in env_update:
+            env_update[_job_id_key] = str(flow_run.id)
+
         new.env.update(env_update)
 
+        # update 'cpu' and 'memory'
+        if flow_run.context.get("cpu"):
+            new.cpu = flow_run.context.get("cpu")
+        if flow_run.context.get("memory"):
+            new.memory = flow_run.context.get("memory")
+
+        # update 'task_customizations'
+        task_customizations_upd = flow_run.context.get("task_customizations", [])
+        new.task_customizations = JsonPatch(
+            new.task_customizations.patch + task_customizations_upd
+        )
```

Comment:
I remember we discussed this, but can you remind me why we had to use json patches?

Reply:
Oh I remember now. Because the agent spawns the jobs. And we want to control the job's network and not the agent's.